public/app/features/dashboard/components/SubMenu/Annotations.tsx

```diff
@@ -1,18 +1,29 @@
 import React, { FunctionComponent, useEffect, useState } from 'react';
 import { LegacyForms } from '@grafana/ui';
+import { EventBusExtended } from '@grafana/data';
+import { CoreEvents } from 'app/types';
 const { Switch } = LegacyForms;
 
 interface Props {
+  events: EventBusExtended;
   annotations: any[];
   onAnnotationChanged: (annotation: any) => void;
 }
 
-export const Annotations: FunctionComponent<Props> = ({ annotations, onAnnotationChanged }) => {
+export const Annotations: FunctionComponent<Props> = ({ events, annotations, onAnnotationChanged }) => {
   const [visibleAnnotations, setVisibleAnnotations] = useState<any>([]);
   useEffect(() => {
     setVisibleAnnotations(annotations.filter(annotation => annotation.hide !== true));
   }, [annotations]);
 
+  useEffect(() => {
+    const handler = () => {
+      setVisibleAnnotations(annotations.filter(annotation => annotation.hide !== true));
+    };
+    events.on(CoreEvents.submenuVisibilityChanged, handler);
+    return () => events.off(CoreEvents.submenuVisibilityChanged, handler);
+  }, []);
```

Comment:
would prefer to not use the bus for this, if can make the annotations array become a new instance when it gets updated (or visibility changed), we can just pass this annotations array to SubMenu and use normal react prop change detection to trigger rendering

Reply:
OK ok, I will look into that during the day. Great feedback, thanks 🙏